### ISSUE TITLE
Data recording stopped after some days of "no reboot"

### ISSUE THREAD

**Issue by rowich** [bassmaster187/TeslaLogger#293]
OK, it happened again, complete day of recording today including charging is missing. Several hundred of kilometers to /dev/null. Admininterface & Grafana working like a charm, but no recording. This seems to append reproducible after some days of "no reboot" (e.g. automatic software update). Don't like to perform a daily reboot (btw: any way to automatise that?)

Raspberry PI 3B, Software v1.44
What data do you require from me to investigate this?

**Comment by superfloh247** [comment]
Please look for something like this in /var/log/messages*

oom_reaper: reaped process 19550 (mono), now anon-rss:0kB, file-rss:0kB, shmem-rss:4kB

**Comment by rowich** [comment]
got it.

Aug 29 22:52:10 raspberry kernel: [781925.371963] Tasks state (memory
values in pages):
Aug 29 22:52:10 raspberry kernel: [781925.371966] [  pid  ]   uid  tgid
total_vm      rss pgtables_bytes swapents oom_score_adj name
Aug 29 22:52:10 raspberry kernel: [781925.371986] [    111]     0  
111     3194       41    22528       59             0 systemd-journal
Aug 29 22:52:10 raspberry kernel: [781925.371992] [    136]     0  
136     3614       14    16384      167         -1000 systemd-udevd
Aug 29 22:52:10 raspberry kernel: [781925.372001] [    271]   100  
271     4319       16    20480       93             0 systemd-timesyn
Aug 29 22:52:10 raspberry kernel: [781925.372007] [    329]   105  
329     1629       54    14336       56          -900 dbus-daemon
Aug 29 22:52:10 raspberry kernel: [781925.372013] [    332]     0  
332     5939       94    20480       99             0 rsyslogd
Aug 29 22:52:10 raspberry kernel: [781925.372020] [    333]     0  
333     1330       16    14336       43             0 cron
Aug 29 22:52:10 raspberry kernel: [781925.372026] [    335]   108  
335     1964      393    16384       61             0 avahi-daemon
Aug 29 22:52:10 raspberry kernel: [781925.372032] [    337]     0  
337     1728       26    16384       64             0 systemd-logind
Aug 29 22:52:10 raspberry kernel: [781925.372038] [    340]     0  
340     2500       12    16384       83             0 wpa_supplicant
Aug 29 22:52:10 raspberry kernel: [781925.372044] [    344] 65534  
344     1324        4    14336       55             0 thd
Aug 29 22:52:10 raspberry kernel: [781925.372051] [    357]   108  
357     1599        6    14336       69             0 avahi-daemon
Aug 29 22:52:10 raspberry kernel: [781925.372057] [    421]     0  
421     2617       16    18432      113             0 wpa_supplicant
Aug 29 22:52:10 raspberry kernel: [781925.372063] [    486]     0  
486      734       37    12288       64             0 dhcpcd
*Aug 29 22:52:10 raspberry kernel: [781925.372069] [    496]     0  
496   212635   180572  1132544     2049             0 mono**
*Aug 29 22:52:10 raspberry kernel: [781925.372075] [    509]     0  
509     1046        0    14336       32             0 agetty
Aug 29 22:52:10 raspberry kernel: [781925.372082] [    518]     0  
518     2552        0    20480      134         -1000 sshd
Aug 29 22:52:10 raspberry kernel: [781925.372088] [    580]   109  
580   190069    26743   344064    19568             0 mysqld
Aug 29 22:52:10 raspberry kernel: [781925.372094] [    584]     0  
584    30311      692    53248       82             0 apache2
Aug 29 22:52:10 raspberry kernel: [781925.372101] [    592]     0  
592      521        1     8192       31             0 hciattach
Aug 29 22:52:10 raspberry kernel: [781925.372107] [    599]     0  
599     1769        0    14336       73             0 bluetoothd
Aug 29 22:52:10 raspberry kernel: [781925.372113] [    610]     0  
610     6679       76    32768      276             0 nmbd
Aug 29 22:52:10 raspberry kernel: [781925.372119] [    749]     0  
749    10281       65    53248      409             0 smbd
Aug 29 22:52:10 raspberry kernel: [781925.372126] [    750]     0  
750     9677       37    47104      421             0 smbd-notifyd
Aug 29 22:52:10 raspberry kernel: [781925.372131] [    751]     0  
751     9680       41    47104      417             0 cleanupd
Aug 29 22:52:10 raspberry kernel: [781925.372137] [    755]     0  
755    10278       55    51200      419             0 lpqd
Aug 29 22:52:10 raspberry kernel: [781925.372144] [    789]   111  
789   254865     3718   139264      993             0 grafana-server
Aug 29 22:52:10 raspberry kernel: [781925.372151] [  10396]    33
10396    30444      843    47104       79             0 apache2
Aug 29 22:52:10 raspberry kernel: [781925.372157] [  10397]    33
10397    30381      763    47104       79             0 apache2
Aug 29 22:52:10 raspberry kernel: [781925.372163] [  10398]    33
10398    30428      808    47104       79             0 apache2
Aug 29 22:52:10 raspberry kernel: [781925.372169] [  10400]    33
10400    30511      891    47104       72             0 apache2
Aug 29 22:52:10 raspberry kernel: [781925.372176] [  10401]    33
10401    30511      845    47104       72             0 apache2
Aug 29 22:52:10 raspberry kernel: [781925.372183] [  13196]    33
13196    30511      850    47104       72             0 apache2
Aug 29 22:52:10 raspberry kernel: [781925.372190] [  20271]    33
20271    30381      732    47104       79             0 apache2
Aug 29 22:52:10 raspberry kernel: [781925.372196] [  20442]     0
20442    10455      187    53248      335             0 smbd
Aug 29 22:52:10 raspberry kernel: [781925.372205] [  22726]     0
22726      472       20    12288        0             0 sessionclean
Aug 29 22:52:10 raspberry kernel: [781925.372211] [  22728]     0
22728      472       21    10240        0             0 sessionclean
Aug 29 22:52:10 raspberry kernel: [781925.372217] [  22730]     0
22730     2491       21    14336        0             0 sort
Aug 29 22:52:10 raspberry kernel: [781925.372223] [  22731]     0
22731     2491       21    14336        0             0 sort
Aug 29 22:52:10 raspberry kernel: [781925.372229] [  22732]     0
22732      472       20    10240        0             0 sessionclean
Aug 29 22:52:10 raspberry kernel: [781925.372235] [  22742]     0
22742    12907      237    34816        0             0 php7.0
Aug 29 22:52:10 raspberry kernel: [781925.372245] [  22862]     0
22862      472       34    10240        0             0 dhcpcd-run-hook
Aug 29 22:52:10 raspberry kernel: [781925.372251] [  22867]     0
22867      472       31    10240        0             0 resolvconf
Aug 29 22:52:10 raspberry kernel: [781925.372257] [  22878]     0
22878      472       32     8192        0             0 resolvconf
*Aug 29 22:52:10 raspberry kernel: [781925.501521] oom_reaper: reaped
process 496 (mono), now anon-rss:0kB, file-rss:0kB, shmem-rss:4kB**
*Aug 30 06:25:03 raspberry liblogging-stdlog:  [origin
software="rsyslogd" swVersion="8.24.0" x-pid="332"
x-info="http://www.rsyslog.com"] rsyslogd was HUPed

Note 1: car was already on a trip (started Aug 30, 06:15) at Aug 30 06:25
Note 2: found this in "messages.1", the last lines are visible,
"messages" starts exactly after the shown content.

Next step?


Am 30/08/2020 um 14:20 schrieb superfloh247:

**Comment by bassmaster187** [comment]
Did you update your system with apt-get update & upgrade?

Could you please post versions of Mono, Grafana, OS and DB listed in logfile after reboot.

**Comment by rowich** [comment]
Yes, I did, but months ago and it was working without issues since
approx 4 weeks ago.

Here is the log content:

29.08.2020 12:38:18 : change TeslaLogger *state: Start -> Sleep*
29.08.2020 12:38:18 : Distance: 6.95257530037006 - Radius: 200 -
Home@Münchenstein
29.08.2020 12:38:18 : Reverse geocoding by Geofence
29.08.2020 12:38:31 : *** Check new Version ***
29.08.2020 12:48:52 : HTTP getchargingstate
29.08.2020 12:51:50 : SetCost
29.08.2020 12:51:50 : JSON:
{"id":392,"cost_currency":"EUR","cost_per_kwh":"0.283","cost_per_session":"","cost_per_minute":"","cost_idle_fee_total":"","cost_kwh_meter_invoice":"30.4","cost_total":"8.60"}
29.08.2020 12:51:50 : SetCost OK: 1
29.08.2020 14:38:44 : *** Check new Version ***
29.08.2020 16:38:58 : *** Check new Version ***
29.08.2020 18:39:09 : *** Check new Version ***
29.08.2020 20:39:27 : *** Check new Version ***
29.08.2020 20:44:28 : IsOnline: Connection reset by peer
30.08.2020 13:56:45 : *Reboot request!*
30.08.2020 13:56:45 : --------------------------------------------
(private address insertion here)
30.08.2020 13:57:04 : Teslalogger Online Version: 1.44.0.0
30.08.2020 13:57:04 : Logfile Version: 1.0.10.0
30.08.2020 13:57:04 : SRTM Version: 0.0.3.0
30.08.2020 13:57:04 : Docker: NO!
30.08.2020 13:57:04 : Current Culture: en-US
30.08.2020 13:57:04 : Mono Runtime: 6.8.0.105 (tarball Tue Feb 4
22:00:05 UTC 2020)
30.08.2020 13:57:04 : Grafana Version: 6.3.5
30.08.2020 13:57:04 : OS Version: RPI3B+ 1.3 / 4.19.66.7
30.08.2020 13:57:06 : Update Settings: all
30.08.2020 13:57:06 : DBConnectionstring:
Server=127.0.0.1;Database=teslalogger;Uid=root;Password=teslalogger;CharSet=utf8;
30.08.2020 13:57:06 : Car#:0
30.08.2020 13:57:06 : KeepOnlineMinAfterUsage: 15
30.08.2020 13:57:06 : SuspendAPIMinutes: 21
30.08.2020 13:57:06 : SleepPositions: 5000
30.08.2020 13:57:06 : UseScanMyTesla: False
30.08.2020 13:57:07 : DB Version: 10.1.44-MariaDB-0+deb9u1
30.08.2020 13:57:07 : Count Pos: 278451
30.08.2020 13:57:08 : Restore Token OK. Age: 8/21/2020 1:18:53 AM

anything missing?

BTW: the JSON comments are to define default charging costs???


Am 31/08/2020 um 08:13 schrieb Christian P.:

**Comment by rowich** [comment]
ok, new behavior with 1.44.9: does not record a drive even after reboot. Implemented a cron job to reboot every night a 04:04, but simply ignored the last ride at 08:30 today. Downgrade mono runtime now? Anything else I could do or deliver to analyse?

**Comment by Scotty1928** [comment]
I have the same issue. No recorded drives since 09/09, even though there have been ~500km ☹️

**Comment by superfloh247** [comment]
Please attach nohup.out to this issue

**Comment by rowich** [comment]
[nohup.zip](https://github.com/bassmaster187/TeslaLogger/files/5214095/nohup.zip)
Note: another 2 rides earlier today (~09:00/10:00) not recorded, last ride is still on Sep-10.

**Comment by bassmaster187** [comment]
@rowich 
You changed your tesla password and there was a bug in change password. 
After an update your car should login to the tesla server. Sorry for this bug.

**Comment by bassmaster187** [comment]
@Scotty1928  did you change your password as well? Or do you have any other problems? Please provide your logfile (nohup.out)

**Comment by Scotty1928** [comment]
Last password change has been roughly four months ago, idk if that's relevant. Will provide the file as soon as i get home (and if i don't forget...)

**Comment by Scotty1928** [comment]
Here are the logs. I included all logs since it stopped working. Hope, that helps.
[nohup.txt](https://github.com/bassmaster187/TeslaLogger/files/5215269/nohup.txt)

**Comment by rowich** [comment]
@bassmaster187 - I did not changed the password since using Teslalogger in March 2019. But anyway, after updating to 1.44.10.0, it's getting current car data (SoC, odometer,...). Let's see whats happening with the next ride.

**Comment by sunfreak** [comment]
Hi, same here. Found Teslalogger stopped data recording on 12.09.2020 06:24 (from Mothership view)! Did an test drive with some visitors yesterday, not recorded.
But no 'oom_reaper' messages on any messages file. Tesla.com reachable:
> root@teslalogger:~# zgrep 'oom_reaper: reaped' /var/log/messages*
> root@teslalogger:~# ping www.tesla.com
> PING e1792.dscx.akamaiedge.net (92.122.25.146) 56(84) bytes of data.
> 64 bytes from a92-122-25-146.deploy.static.akamaitechnologies.com (92.122.25.146): icmp_seq=1 ttl=60 time=13.9 ms
> 64 bytes from a92-122-25-146.deploy.static.akamaitechnologies.com (92.122.25.146): icmp_seq=2 ttl=60 time=13.8 ms
> 64 bytes from a92-122-25-146.deploy.static.akamaitechnologies.com (92.122.25.146): icmp_seq=3 ttl=60 time=13.7 ms
> 64 bytes from a92-122-25-146.deploy.static.akamaitechnologies.com (92.122.25.146): icmp_seq=4 ttl=60 time=13.7 ms
> 64 bytes from a92-122-25-146.deploy.static.akamaitechnologies.com (92.122.25.146): icmp_seq=5 ttl=60 time=13.7 ms
> 64 bytes from a92-122-25-146.deploy.static.akamaitechnologies.com (92.122.25.146): icmp_seq=6 ttl=60 time=13.7 ms
> 64 bytes from a92-122-25-146.deploy.static.akamaitechnologies.com (92.122.25.146): icmp_seq=7 ttl=60 time=13.7 ms

I don't reboot out of crontab, but until now only on demand. Currently my TL is still in this state (version 1.44.9.0 showing update to 1.44.10.0 available).

Any hint to analyse the issue in this state? @bassmaster187 I can also give you ssh access.
![image](https://user-images.githubusercontent.com/3902974/93056567-e4939700-f66c-11ea-9aef-6b4fb3b8ef91.png)

**Comment by rowich** [comment]
OK, what I did (probably at the evening of the 10th (after successful recording a charge in the hotel, but not the drive afterwards): I reentered my password into the new dialog supporting multiple car - but still the same password. So not real password change, but reenter.

Good: Last drive today recorded successfully with 1.44.10.0. If the next bigger trip on Wednesday and Thursday gets recorded without any hassle I am going to close the issue from my point of view. Any other opinions from @sunfreak and @Scotty1928 ?
